jack-core/src/com/rapleaf/jack/transaction/DbMetricsImpl.java

```diff
@@ -0,0 +1,157 @@
+
+      maxConnectionWaitTime = connectionPool.getMaxBorrowWaitTimeMillis();
+      totalActiveTime += numActive * updateToNowTime;
+      totalIdleTime += numIdle * updateToNowTime;
+      totalConnectionWaitTime += numWaiters * updateToNowTime;
+
+      if (isOpenConnection) {
+        totalConnections += 1;
+      }
+      int newConnectionsNumber = numActive + numIdle;
+      if (newConnectionsNumber > currentConnectionsNumber) {
+        openedConnectionsNumber += newConnectionsNumber - currentConnectionsNumber;
+      }
+      currentConnectionsNumber = newConnectionsNumber;
+
+
+      if (numActive == maxTotalConnections) {
+        maxActiveConnectionsTime += updateToNowTime;
+      }
+
+      if ((numIdle + numActive == minIdleConnections) && (numIdle > 0)) {
+        minIdleConnectionsTime += updateToNowTime;
+      }
+    } catch (Exception e) {
+      LOG.error("failed to update statistics", e);
+    }
+  }
+
+  @Override
+  public long getTotalConnections() {
+    return totalConnections;
+  }
+
+  @Override
+  public double getMaxConnectionsProportion() {
+    return (double)maxActiveConnectionsTime / (double)lifeTimeStopwatch.elapsedMillis();
+  }
+
+  @Override
+  public double getAverageConnectionWaitTime() {
+    return (double)totalConnectionWaitTime / (double)totalConnections;
+  }
+
+  @Override
+  public double getAverageConnectionExecutionTime() {
+    return ((double)totalActiveTime / (double)totalConnections);
+  }
+
+  @Override
+  public double getMinConnectionsProportion() {
+    return (double)minIdleConnectionsTime / (double)lifeTimeStopwatch.elapsedMillis();
+  }
+
+  @Override
+  public long getOpenedConnectionsNumber() {
+    return openedConnectionsNumber;
+  }
+
+  @Override
+  public double getAverageIdleConnections() {
+    return (double)totalIdleTime / (double)lifeTimeStopwatch.elapsedMillis();
+  }
+
+  @Override
+  public double getAverageActiveConnections() {
+    return (double)totalActiveTime / (double)lifeTimeStopwatch.elapsedMillis();
+  }
+
+  @Override
+  public long getMaxConnectionWaitingTime() {
+    return maxConnectionWaitTime;
+  }
+
+  @Override
+  public String log() {
+    String log = "";
+
+    log += ("\n-----------------------TRANSACTOR METRICS-----------------------\n");
+    log += ("\nAverage (over time) number of Idle connections : " + String.valueOf(getAverageIdleConnections()));
+    log += ("\nAverage (over time) number of Active connections : " + String.valueOf(getAverageActiveConnections()));
+    log += ("\nTotal number of querys/executions : " + String.valueOf(getTotalConnections()));
+    log += ("\nConnections opened : " + String.valueOf(getOpenedConnectionsNumber()));
+    log += ("\nProportion of time when the number of connections is maximum : " + String.valueOf
+        (getMaxConnectionsProportion()));
+    log += ("\nProportion of time when all connections are Idle but not killed : " + String.valueOf
+        (getMinConnectionsProportion()));
```

Comment:
The description of these two metrics is kind of long, and not very straightforward. What about changing to something like `Max capacity time (%)` and `All idle time (%)`?